src/Microsoft.OData.Core/Json/JsonWriterAsync.cs

```diff
@@ -317,24 +317,25 @@ public Task EndTextWriterValueScopeAsync()
         /// Asynchronously writes a separator of a value if it's needed for the next value to be written.
         /// </summary>
         /// <returns>A task that represents the asynchronous write operation.</returns>
-        private async Task WriteValueSeparatorAsync()
+        private Task WriteValueSeparatorAsync()
         {
             if (this.scopes.Count == 0)
             {
-                return;
+                return TaskUtils.CompletedTask;
             }
 
             Scope currentScope = this.scopes.Peek();
             if (currentScope.Type == ScopeType.Array)
             {
-                if (currentScope.ObjectCount != 0)
+	            currentScope.ObjectCount++;
+                if (currentScope.ObjectCount > 1)

```

Comment:
@henning-krause I initially had problems wrapping my head around moving the line that increments `ObjectCount` about the if statement together with the change in the conditional expression so the team took time to test it out and we confirmed your logic is solid. Either of these 3 are logically sound:
```csharp
currentScope.ObjectCount++;
if (currentScope.ObjectCount > 1)
{
    // return ...
}

// Or

if (currentScope.ObjectCount++ != 0)
{
    // return ...
}

// Or

if (currentScope.ObjectCount++ > 0)
{
    // return ...
}
```
The majority had no problem with the post-increment version. I think your version is easier to read/understand only as far as the incrementing statement. Otherwise the conditional expression takes a bit of time to follow. However, we are okay with your version if you're convinced it's easier to follow.

Reply:
FYI, I would have expected:

if (currentScope.ObjectCount++ > 0)
{
    // return ...
}